### PR TITLE
Simplify the clustering example

### PR DESCRIPTION
**Pull request by benkli01**
- The trainable parameters were reduced (from 3,274,698 to 20,410) to
  - reduce over-parameterization,
  - make the example faster to run and
  - make it easier to visualize the clustering results
- Some re-factoring to improve readability

### CODE REVIEW

File: tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py [hunk 1]
```diff
@@ -41,145 +41,141 @@
...
+  score = clustered_model.evaluate(x_test, y_test, verbose=0)
+  print('Clustered model test loss:', score[0])
+  print('Clustered model test accuracy:', score[1])
+
+  return clustered_model
+
+
+def test_clustered_model(clustered_model, x_test, y_test):
+  # Ensure accuracy persists after deserializing the model
+  clustered_model.compile(
```
Comment (akarmi) [inline tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py:131]: Why do we need to compile the model here?

Comment (benkli01) [reply tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py:131]: Good point, I think we don't! I will remove this.

File: tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py [hunk 2]
```diff
@@ -41,145 +41,136 @@
...
+
+  # Model needs to be clustered after initial training
+  # and having achieved good accuracy
+  model.fit(
+      x_train,
+      y_train,
+      batch_size=batch_size,
+      epochs=epochs,
+      verbose=1,
+      validation_data=(x_test, y_test))
```
Comment (Ruomei) [inline tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py:85]: Should we use validation_split instead?

**Comment by Ruomei** [comment]
LGTM and I left a minor suggestion.

File: tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py (hunk 2 above)
Comment (benkli01) [reply tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py:85]: Yes, that makes sense.

Comment (benkli01) [reply tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py]: I have changed both occurrences. Thanks.

**Review by Ruomei — APPROVED** [review]
LGTM.

**Review by akarmi — APPROVED** [review]

File: tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py [hunk 3]
```diff
@@ -41,145 +41,137 @@
...
+    'number_of_clusters': 8,
+    'cluster_centroids_init': cluster_config.CentroidInitialization.DENSITY_BASED
+  }
+
+  # Cluster model
+  clustered_model = cluster.cluster_weights(model, **clustering_params)
+
+  # Use smaller learning rate for fine-tuning
+  # clustered model
+  opt = tf.keras.optimizers.Adam(learning_rate=1e-5)
```
Comment (wwwind) [inline tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py:107]: i've got better accuracy with learning rate 1e-3
 0.9724 (1e-5) vs. 0.9786 (1e-3) 
could be noise, but it trains faster with 1e-3
